model/patch/db.go

```diff
@@ -177,14 +178,23 @@ func PatchesByProject(projectId string, ts time.Time, limit int) db.Q {
 	}).Sort([]string{"-" + CreateTimeKey}).Limit(limit)
 }
 
+// FindFailedCommitQueuePatchesInTimeRange returns failed patches if they started within range,
+// or if they were never started but finished within time range. (i.e. timed out)
 func FindFailedCommitQueuePatchesinTimeRange(projectID string, startTime, endTime time.Time) ([]Patch, error) {
 	query := bson.M{
 		ProjectKey: projectID,
 		StatusKey:  evergreen.PatchFailed,
 		AliasKey:   evergreen.CommitQueueAlias,
-		"$and": []bson.M{
-			{StartTimeKey: bson.M{"$gte": startTime}},
-			{StartTimeKey: bson.M{"$lte": endTime}},
+		"$or": []bson.M{
+			{"$and": []bson.M{
+				{StartTimeKey: bson.M{"$lte": endTime}},
+				{StartTimeKey: bson.M{"$gte": startTime}},
+			}},
+			{"$and": []bson.M{
+				{StartTimeKey: util.ZeroTime},
```

Comment:
if StartTime isn't set I think it will equal Go's zero time (Jan 1 0001, ...)

Reply:
Sorry I'm not sure I understand this comment (util.ZeroTime is Go's zero time).

Reply:
My understanding is that util.Zero time is Jan 1 1970 (the epoch), while Go's time.Time{} is Jan 1 0001
https://github.com/evergreen-ci/evergreen/blob/aa0ca38f8d19aa98ca78d83169e1fd3d03ac6c41/util/time.go#L9

Reply:
Oh my mistake!